subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -289,6 +295,60 @@ Gradle will treat these directories as possible installations but will not desce
 org.gradle.java.installations.paths=/custom/path/jdk1.8,/shared/jre11
 ----
 
+[NOTE]
+====
+Custom toolchains are not prioritized over <<sec:auto_detection,auto-detected>> ones. If auto-detection is turned on, custom toolchains
+can still be specified but all they will do is to extend the set of detected JRE/JDK locations. Among these locations, one will be picked
+according to the <<sec:precedence,Toolchain Precedence Rules>>.
+====
+
+[[sec:precedence]]
+== Precedence
+
+When there are multiple local JRE/JDK installations matching the toolchain specification of a build, one will be picked according to the
+following rules:
+
+. installation currently running Gradle is preferred over any other
+. JDK locations are preferred over JRE ones
+. certain vendors take precedence over others; their ordering (from the highest priority to lowest):
+.. ADOPTIUM
+.. ADOPTOPENJDK
+.. AMAZON
+.. APPLE
+.. AZUL
+.. BELLSOFT
+.. GRAAL_VM
+.. HEWLETT_PACKARD
+.. IBM
+.. IBM_SEMERU
+.. MICROSOFT
+.. ORACLE
+.. SAP
+.. everything else
+. higher major versions take precedence over lower ones
+. higher minor versions take precedence over lower ones
+. installation paths take precedence according to their lexicographic ordering (last resort criteria for deterministically deciding
+between installations of the same type, from the same vendor and with the same version)
+
+All these rules are applied as multilevel sorting criteria. Let's illustrate with an example. Let's say that the only
+criteria we have added to our toolchain specification is that we are interested in HotSpot implementations and that on our local machine
+following installations are detected: *Oracle JRE v17.0.1*, *Oracle JDK v17.0.0*, *Microsoft JDK 11.0.13*, *Microsoft JDK 11.0.12.7.1* and
+AdoptOpenJDK 16.0.1.j9, which is running Gradle, but which doesn't match the toolchain specification.
+
+The first rule is applied, but since the installation running Gradle doesn't match the toolchain specification (it's not a HotSpot
+JVM), this rule doesn't yield anything, so we continue. If this wouldn't be the case, then the installation running Gradle would be picked,
+and we would be done.
+
+The second rule is applied, ie. we prefer JDK installations, so we are left with 3 matching ones: *Oracle JDK v17.0.0*, *Microsoft JDK 11.0.13*, *Microsoft JDK 11.0.12.7.1*.
+
+The third rule is applied, ie. we prefer Microsoft as a vendor over Oracle, so we are left with 2 of them: *Microsoft JDK 11.0.13*,
+*Microsoft JDK 11.0.12.7.1*.
+
+Then the fourth rule is applied, the highest version number wins. The major version is the same, but the minor versions aren't, so
+*Microsoft JDK 11.0.13* wins.
+
+Since we have narrowed down the list to a single candidate, no further rules need to be applied.
```

Comment:
I think it is great to have an example here 👍 
However, the way it is currently phrased gives the wrong impression IMO.
I would instead present the different options as you did, then show them sorted, then explain, in order, why the first ones are rejected until we get to the selected one.

Reply:
The thing is that the sorting we do is not relevant conceptually. It's just an implementation detail. We could have implemented the "pick one from the many" in a completely different way, without any sorting and the documentation should still be the same. Btw. not even the implementation works like that, it's not "sort, then go through it", it's just "sort, and whatever ends up first wins" (see [here](https://github.com/gradle/gradle/blob/af58e97110e26c02eadb57eda757c8722b8b0893/subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java#L82)).

I will try to find a form that makes us both happy(er).

Reply:
Thanks for the pointer, I had the wrong model. So we effectively sort valid candidates and take the first. Let's then use:
* Present the different options
* Discard those that do not match
* Sort the remaining ones